### ISSUE TITLE
epoch results are much better than test results

### ISSUE THREAD

**Issue by DilaraAlbayrak** [CZHQuality/Sal-CFS-GAN#4]
Hi,
Thanks for the source. I have trained GazeGAN_using_CSC for 106 epochs, and train results seem pretty good. I am sharing one synthesized map, and its corresponding ground-truth to demonstrate the similarity. However, when I test with that training, I couldn't get that successful results, how can I get very similar results during the test, either?

![epoch093_real_image](https://user-images.githubusercontent.com/7215154/73139635-0231d300-4081-11ea-8e22-1cde772938ca.jpg)
![epoch093_synthesized_image](https://user-images.githubusercontent.com/7215154/73139636-0231d300-4081-11ea-80dc-67be49681e4b.jpg)

**Comment by CZHQuality** [comment]
Hi DilaraAlbayrak,

Thanks for your attention and patience.

To my experience, I think the most suspicious reason for your concern is the Over-Fitting.

In my implementation, I only trained the model for around 20-30 epoches (maximum), and the performance on the test set is pretty good, looks very similar to the training performance.

Considering that you have trained your model for around 100 epoches, I guess the model has already over-fitted the training set, thus causing very poor generalization performance on the unseen test samples.

Technically speaking, please decrease the number of epoches, and make sure to exploit a validation set (select some images from the test set) to monitor the generalization performance of your model after each training epoch, then select the version/epoch that obtaining the best generalization performance.

Notably, If you adopt two totally different datasets as training and testing sets respectively, such as using SALICON as training set while using MIT1003 as testing set, please select some images from MIT1003 to fine-tune the model, because there is a quite huge domain-shift between two different datasets.

If you have any question, please don't hesitate to contact with me.

**Comment by DilaraAlbayrak** [comment]
Thank you I will correct my training.

**Comment by DilaraAlbayrak** [comment]
@CZHQuality   while I am training is validation also conducted at the end of each epoch? Or should I add something else? Thanks.
